proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,456 @@
+
+
+## Managing the existing backlog
+
+As of 2021-12-15, there are 188 pull requests and 436 issues in the AWS CLI
+GitHub repository. 43 pull requests and 87 issues have not been updated in more
+than a year. 95 issues have ten or more upvotes. 
+<!-- [TBD] open pull requests are directly related to an existing feature request. -->
+
+Based on these numbers, we propose to manage existing pull requests in two
+phases. First, we will determine if there is a corresponding tracking issue for
+each pull request and [link them
+together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
+If no tracking issue exists, we will create one and link it to the pull request.
+Then, we will review all open issues and apply the new specification to identify
+the current stage. We will prioritize the review of issues with a linked pull
+request. After review, all pull requests will follow the process and timelines
+defined in the specification.
+
+## Rationale/FAQ
+
+### Q. Why do we limit the number of in flight reviews?
+
+We do not want to discourage users from contributing or proposing changes, but
+we need to acknowledge that the maintainers have limited bandwidth to give
+feedback on implementation or design and review pull requests. The maintainers
+have work that comes from outside of the GitHub repository that needs to be
+prioritized along with user contributions. Controlling the volume of pull
+request reviews and prioritizing completing the in flight reviews over new
+reviews will result in a shorter time to merge and release a contributor's
+change for a feature requested by the community. It will also reduce the amount
+of context switching for the maintainers, allowing them to focus on giving high
+quality feedback to users. As we use this framework to process existing feature
+requests, we can revisit what the in flight limit should be.
+
+### Q. Why do we require issues to be opened instead of just pull requests?
+
+Having a defined process for intake reduces the burden on deciding which way is
+the best way. We're taking an opinionated stance that code changes made in
+isolation generally do not provide enough context on what problem exists and why
+the proposed change is the best way to fix it. Reviewing a pull request also
+requires significant effort on the part of maintainers - reading someone elses
+code can be an intensive task. Before dedicating that effort, we feel that
+having a more general discussion about the problem and solutions will help
+reduce the overall effort into code review.
+
+Pull requests are often made for problems that only affect the contributor or a
+very small portion of the user base. Requiring an issue provides a mechanism to
+request structured feedback in the form of "upvotes" or GitHub reactions to
+estimate the impact of the issue on the community.
+
+Logistically, using GitHub issues also centralizes the metadata on what work is
```

Comment:
I'm not sure how strong of a reason this is from the perspective of the community. I'd lean toward removing it. Plus another point that would be good to capture is:

It solidifies the position that all PRs that are currently open are something we'd like to pull in. It removes any ambiguity for a contributor that has an open PR on whether progress has not been made on it because we are not going to accept it or we just have not gotten around to review it.